src/battle_ai_switch_items.c

```diff
@@ -897,37 +1079,774 @@ static u32 GetBestMonDmg(struct Pokemon *party, int firstId, int lastId, u8 inva
+            {
+                tSpikesDamage = maxHP / 16;
+                if (tSpikesDamage == 0)
+                    tSpikesDamage = 1;
+            }
+            hazardDamage += tSpikesDamage;
+        }
+    }
+    return hazardDamage;
+}
+
+// Gets damage / healing from weather
+static s32 GetSwitchinWeatherImpact(void)
+{
+    s32 weatherImpact = 0, maxHP = AI_DATA->switchinCandidate.battlemon.maxHP, ability = AI_DATA->switchinCandidate.battlemon.ability;
+    u16 item = AI_DATA->switchinCandidate.battlemon.item;
+
+    if (WEATHER_HAS_EFFECT)
+    {
+        // Damage
+        if (item != ITEM_SAFETY_GOGGLES)
+        {
+            if ((gBattleWeather & B_WEATHER_HAIL) && (AI_DATA->switchinCandidate.battlemon.type1 != TYPE_ICE || AI_DATA->switchinCandidate.battlemon.type2 != TYPE_ICE)
+            && ability != ABILITY_OVERCOAT && ability != ABILITY_SNOW_CLOAK && ability != ABILITY_ICE_BODY)
+            {
+                weatherImpact = maxHP / 16;
+                if (weatherImpact == 0)
+                    weatherImpact = 1;
+            }
+            else if ((gBattleWeather & B_WEATHER_SANDSTORM) && (AI_DATA->switchinCandidate.battlemon.type1 != TYPE_GROUND && AI_DATA->switchinCandidate.battlemon.type2 != TYPE_GROUND
+                && AI_DATA->switchinCandidate.battlemon.type1 != TYPE_ROCK && AI_DATA->switchinCandidate.battlemon.type2 != TYPE_ROCK
+                && AI_DATA->switchinCandidate.battlemon.type1 != TYPE_STEEL && AI_DATA->switchinCandidate.battlemon.type2 != TYPE_STEEL
+                && ability != ABILITY_OVERCOAT && ability != ABILITY_SAND_VEIL && ability != ABILITY_SAND_RUSH && ability != ABILITY_SAND_FORCE))
+            {
+                weatherImpact = maxHP / 16;
+                if (weatherImpact == 0)
+                    weatherImpact = 1;
+            }
+        }
+        if ((gBattleWeather & B_WEATHER_SUN) && (ability == ABILITY_SOLAR_POWER || ability == ABILITY_DRY_SKIN))
+        {
+            weatherImpact = maxHP / 8;
+            if (weatherImpact == 0)
+                weatherImpact = 1;
+        }
+
+        // Healing
+        if (gBattleWeather & B_WEATHER_RAIN)
+        {
+            if (ability == ABILITY_DRY_SKIN)
+            {
+                weatherImpact = maxHP / 8;
+                if (weatherImpact == 0)
+                    weatherImpact = 1;
+            }
+            else if (ability == ABILITY_RAIN_DISH)
+            {
+                weatherImpact = maxHP / 16;
+                if (weatherImpact == 0)
+                    weatherImpact = 1;
+            }
+        }
+        if (((gBattleWeather & B_WEATHER_HAIL) || (gBattleWeather & B_WEATHER_SNOW)) && ability == ABILITY_ICE_BODY)
+        {
+            weatherImpact = maxHP / 16;
+            if (weatherImpact == 0)
+                weatherImpact =1;
+        }
+    }
+    return weatherImpact;
+}
+
+// Gets one turn of recurring healing
+static u32 GetSwitchinRecurringHealing()
```

Comment:
`static u32 GetSwitchinRecurringHealing(void);`

Because it doesn't take any arguments.
`()` means the function can take any number of arguments.